model/btp_application_certificate.py

```diff
@@ -0,0 +1,90 @@
+from model.base import (
+    NamedModelElement,
+)
+
+
+class BtpApplicationCertificate(NamedModelElement):
+    def subject_template(self) -> str:
+        '''
+        certificate subject template to inject CN
+        '''
+        return self.raw['subject_template']
+
+    def common_name(self) -> str:
+        '''
+        subject common name
+        '''
+        return self.raw['common_name']
+
+    def scopes(self) -> list[str]:
+        '''
+        scopes for certificate
+        '''
+        return self.raw['scopes']
+
+    def subject(self, cn: str) -> str:
+        return self.subject_template().format(cn=cn)
+
+    @staticmethod
+    def get_serial_no_from_common_name(cn: str) -> tuple[int, str]:
+        idx = cn.find(".")
+        if idx <= 0:
+            return 0, cn
+        try:
+            return int(cn[:idx]), cn[idx+1:]
+        except ValueError:
+            return 0, cn
```

Comment:
why not simply let the exception propagate, and have the callers deal w/ it? currently, callers all seem to check for `0`, and then raise themselves..

Reply:
done